torchinductor/codecache.py

```diff
@@ -63,15 +74,18 @@ def cpp_compiler():
 
 @functools.lru_cache(1)
 def cpp_compiler_search(search):
-    for cxx in search:
-        try:
-            if cxx is None:
-                cxx = install_gcc_via_conda()
-            subprocess.check_output([cxx, "--version"])
-            return cxx
-        except (subprocess.SubprocessError, FileNotFoundError):
-            continue
-    raise exc.InvalidCxxCompiler()
+    lock_dir = get_lock_dir()
+    lock = FileLock(os.path.join(lock_dir, "g++.lock"), timeout=LOCK_TIMEOUT)
+    with lock:
```

Comment:
I think the lock should only be needed around `install_gcc_via_conda()`.  If we are using an existing compiler, it should be fine to test it in parallel.

Reply:
Yes, make sense. I will move the lock before `install_gcc_via_conda()`.